docker/dev_env/Dockerfile

```diff
@@ -0,0 +1,46 @@
+# Use arch. Alpine is nice and small, but permissions and user management
+# are obscure, easy to break, and way too tedious to maintain for a development
+# environment that will inevitably have continuously changing requirements
+FROM docker.io/library/archlinux:latest
+
+# Dependency explanations
+# - git: required by some python package; contributors should use git on their host
+# - make: required by some pre-commit hooks for node-gyp
+# - perl: used in linting
+# - bash: used for scripts
+# - python: language runtime
+# - python-pipx: python CLI app installer
+# - docker*: used to interact with host docker socket
+# - nodejs, npm: language runtime and dependency manager (system npm required by node-based pre-commit hooks)
+# - just: command runner
+# - n: nodejs distribution manager
```

Comment:
`n` is something I have not used before. I am always used `nvm` for managing Node versions. I don't have any specific reasons to favour one over the other, but just noting it down as something that differs from my personal toolchain.

Reply:
I don't know why, but `nvm` (and relatedly `pyenv`) have always driven me to tears of frustration with how they work... and `n` just does the thing I expect it to do every time. `nvm` for some reason _always_ wants to compile Node from scratch on my computer! `pyenv` does the same thing for some reason? I think I tried `asdf` and had a similar thing happen with Python, Node, or something else. `nvm`'s magic of wanting to auto-detect and change Node versions for you was endlessly frustrating... I guess I have had a bad experience with it :sweat_smile: Some of that might come from ignorance or may have been fixed in the many years since I've bothered with it.

I don't know, happy to use whatever here really, but in my recent experience `n` always downloads the binary distribution and "just works" (including `n install auto`, which I never figured out with nvm). If `nvm` works better than `n` for some reason, happy to switch to it. With this `ov`/docker based approach, it theoretically shouldn't matter, as long as someone gets it working right it should "just work" for everyone.